frontend/src/components/common/SimpleTable.tsx

```diff
@@ -77,30 +81,55 @@ export default function SimpleTable(props: SimpleTableProps) {
   }
 
   React.useEffect(() => {
-    setPage(0);
+    if (currentData === data) {
+      if (page !== 0) {
+        setPage(0);
+      }
+      return;
+    }
+
+    // If the currentData is not up to date and we are in the first page, then update
+    // it directly. Otherwise it will require user's intervention.
+    if ((!currentData || currentData.length === 0) || page === 0) {
+      setCurrentData(data);
+    }
   },
-  [data]);
+  // eslint-disable-next-line
+  [data, currentData]);
 
   function getPagedRows() {
     const startIndex = page * rowsPerPage;
     return filteredData.slice(startIndex, startIndex + rowsPerPage);
   }
 
-  if (data === null) {
+  if (currentData === null) {
     return <Loader />;
   }
 
-  let filteredData = data;
+  let filteredData = currentData;
 
   if (filterFunction) {
-    filteredData = data.filter(filterFunction);
+    filteredData = currentData.filter(filterFunction);
   }
 
   return (
-    (!data || data.length === 0) ?
+    (!currentData || currentData.length === 0) ?
       <Empty>{emptyMessage || 'No data to be shown.'}</Empty>
       :
       <React.Fragment>
+        { // Show a refresh button is the data is not up to date, so we allow the user to keep
```

Comment:
A tiny tiny comment fix
```suggestion
        { // Show a refresh button **if** the data is not up to date, so we allow the user to keep
```